crates/keys/src/tests/mod.rs

```diff
@@ -350,4 +350,22 @@ mod key_tests {
         let result = VerifyingKey::try_from(encoded);
         assert!(result.is_err());
     }
+
+    #[test]
+    fn test_verifying_key_with_invalid_bytes() {
+        let invalid_bytes = [0u8; 1];
+
+        let algorithms = [
+            CryptoAlgorithm::Ed25519,
+            CryptoAlgorithm::Secp256k1,
+            CryptoAlgorithm::Secp256r1,
+            CryptoAlgorithm::Eip191,
+            CryptoAlgorithm::CosmosAdr36,
+        ];
+
+        for algorithm in algorithms {
+            let result = VerifyingKey::from_algorithm_and_bytes(algorithm, &invalid_bytes);
+            assert!(result.is_err());
```

Comment:
It would be nicer if you could use `matches!()` and the actual error its supposed to emit